packages/store/addon/-private/ts-interfaces/utils.ts

```diff
@@ -2,4 +2,4 @@
   @module @ember-data/store
 */
 
-export type Dict<K extends string, V> = { [KK in K]: V };
+export type Dict<V> = { [key: string]: V };
```

Comment:
This definition of `Dict` describes an object where we're 100% sure we'll find present values for any possible property key.

It turns out that this is usually a more accurate type, which puts responsibility on the consumer to check whether they got something after retrieving a property.
```suggestion
export type Dict<V> = { [key: string]: V | undefined };
```

There are some cases that I've discussed with @igorT where it may be more reasonable to assume the properties are definitely present (i.e., `Object.keys(myObj).forEach` ) but I would argue that the more common case is that you _want_ undefined here.

Reply:
I agree for now but also agree with @igorT. I wonder if we should have a `StrictDict` for when we don't need to check? Otherwise iteration with `Object.keys` gets very annoying.